src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/SetAzureKeyVaultManagedStorageAccount.cs

```diff
@@ -0,0 +1,101 @@
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.Set, CmdletNoun.AzureKeyVaultManagedStorageAccount,
+        SupportsShouldProcess = true,
+        HelpUri = Constants.KeyVaultHelpUri)]
+    [OutputType(typeof(ManagedStorageAccount))]
+    public class SetAzureKeyVaultManagedStorageAccount : KeyVaultCmdletBase
+    {
+        #region Input Parameter Definitions
+        [Parameter(Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment.")]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Key Vault managed storage account name. Cmdlet constructs the FQDN of a managed storage account name from vault name, currently " +
+                          "selected environment and manged storage account name." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.StorageAccountName )]
+        public string AccountName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure resource id of the storage account." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.StorageAccountResourceId )]
+        public string AccountResourceId { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 3,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the storage account key that must be used for generating sas tokens." )]
+        [ValidateNotNullOrEmpty]
+        public string ActiveKeyName { get; set; }
+
+        [Parameter( Mandatory = false,
+            HelpMessage = "Auto regenerate key. If true, then the managed storage account's inactive key gets auto regenerated and becomes the new active key " +
+                          "after the regeneration period. If false, then the keys of managed storage account are not auto regenerated." )]
+        public SwitchParameter DisableAutoRegenerateKey { get; set; }
+
+        [Parameter( Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Regeneration period. If auto regenerate key is enabled, this value specifies the timespan after which managed storage account's inactive key" +
+                          "gets auto regenerated and becomes the new active key." )]
+        [ValidateNotNull]
+        public TimeSpan? RegenerationPeriod { get; set; }
+
+        [Parameter( Mandatory = false, 
+            HelpMessage = "Disables the use of managed storage account's key for generation of sas tokens." )]
+        public SwitchParameter Disable { get; set; }
```

Comment:
This parameter is named along the lines of similar parameter in Set-AzureKeyVaultSecret and Set-AzureKeyVaultKey and carries similar meaning of disabling use of the key vault object. 

---
In reply to: [119725609](https://github.com/Azure/azure-powershell/pull/4061#discussion_r119725609) [](ancestors = 119725609)